main.go

```diff
@@ -39,6 +39,8 @@ var opts struct {
 	ModulePath string `short:"p" long:"module_path" default:"./vendor/modules" description:"File path to install generated terraform modules, if not overridden by 'destinations:' field"`
 
 	TerrafilePath string `short:"f" long:"terrafile_file" default:"./Terrafile" description:"File path to the Terrafile file"`
+
+	Clean bool `short:"c" long:"clean" description:"Remove all artifacts from destinations and module path upon fetching module(s)"`
```

Comment:
out of curiosity, what is the flag cleaning up that is not already being done in:
1. The git clone [directory](https://github.com/coretech/terrafile/blob/44cf13d1cc571f14f2440395a214eaa2b2526585/main.go#L60)
2. The destinations' [directory](https://github.com/coretech/terrafile/blob/44cf13d1cc571f14f2440395a214eaa2b2526585/main.go#L150)

Reply:
sorry, just noticed https://github.com/coretech/terrafile/pull/13#issuecomment-1431006892, and realized that our cleanup  in 2, is only deleting the `module` folder and not the entire destination folder.

Is it possible in order to make this clear, to add to the cli option description text that you have, a warning that this will delete all files in the destinations including non-modules.?

Reply:
Right, both clean only single module, not entire destination. Following DevOps concepts, destination shall be managed ONLY by terrafile and contain modules which are currently in configuration. Without `--clean` flag, in case module was removed from config, some modules were left in destination(s) while no representation in configuration. Hence, it was confusing and error prone.

Meanwhile, I tried to keep back compatibility intact, so everyone decides on his own strategy to manage dependencies ;-)